Exiled.Events/Patches/Events/Player/ActivatingWarheadPanel.cs

```diff
@@ -7,72 +7,151 @@
-                    __instance.OnInteract();
-                }
-
-                return false;
-            }
-            catch (Exception exception)
+            var newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            // Index offsets.
+            const int offset = 1;
+            const int insideIfOffset = 0;
+            const string permission = "CONT_LVL_3";
+
+            // Search for the eighth "ldarg.0".
+            var index = newInstructions.FindIndex(instruction => instruction.opcode == OpCodes.Stloc_1) + offset;
+
+            // Search for the last "ldloc.0"
+            var insideIfIndex = newInstructions.FindLastIndex(instruction => instruction.opcode == OpCodes.Ldloc_0) + insideIfOffset;
+
+            // Declare List<string> local variable.
+            var list = generator.DeclareLocal(typeof(List<string>));
+
+            // Get the first label of the last "ret".
+            var returnLabel = newInstructions[newInstructions.Count - 1].labels[0];
+
+            // Define the continue label.
+            var continueLabel = generator.DefineLabel();
+
+            // Define the inside if label.
+            var insideIfLabel = newInstructions[insideIfIndex].WithLabels(generator.DefineLabel()).labels[0];
+
+            // Define the checked bypass or contains label.
+            var checkedBypassLabel = generator.DefineLabel();
+
+            // Define the checked contains label label.
+            var checkedContainsLabel = generator.DefineLabel();
+
+            // if (!this._sr.BypassMode && itemById == null)
+            //   return;
+            //
+            // var list = ListPool<string>.Shared.Rent()
+            // var ev = new ActivatingWarheadPanelEventArgs(Player.Get(this.gameObject), list, __instance._sr.BypassMode || itemById.permissions.Contains("CONT_LVL_3"));
+            //
+            // Handlers.Player.OnActivatingWarheadPanel(ev);
+            //
+            // ListPool<string>.Shared.Return(list);
+            //
+            // if (ev.IsAllowed)
+            //   goto insideIfLabel;
+            //
+            // return;
+            newInstructions.InsertRange(index, new[]
             {
-                API.Features.Log.Error($"{typeof(ActivatingWarheadPanel).FullName}.{nameof(Prefix)}:\n{exception}");
+                // if (this._sr.BypassMode)
+                //   goto continueLabel
+                new CodeInstruction(OpCodes.Ldarg_0),
+                new CodeInstruction(OpCodes.Ldfld, Field(typeof(PlayerInteract), nameof(PlayerInteract._sr))),
+                new CodeInstruction(OpCodes.Ldfld, Field(typeof(ServerRoles), nameof(ServerRoles.BypassMode))),
+                new CodeInstruction(OpCodes.Brtrue_S, continueLabel),
+
+                // if (itemByID == null)
+                //   return;
+                new CodeInstruction(OpCodes.Ldloc_1),
+                new CodeInstruction(OpCodes.Brfalse_S, returnLabel),
```

Comment:
Add a continue label and jump to it if `ldloc_1` doesn't push null onto the stack, and add after that `ret` opcode instead of jumping to a return label. Basically, it'll be better for understanding due to the lack of catching a return label, it really isn't necessary.